bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/HandlerThermostatController.java

```diff
@@ -63,17 +77,48 @@ public String[] getSupportedInterface() {
         if (TARGET_SETPOINT.propertyName.equals(property)) {
             return new ChannelInfo[] { TARGET_SETPOINT };
         }
+        if (LOWER_SETPOINT.propertyName.equals(property)) {
+            return new ChannelInfo[] { LOWER_SETPOINT };
+        }
+        if (UPPER_SETPOINT.propertyName.equals(property)) {
+            return new ChannelInfo[] { UPPER_SETPOINT };
+        }
+        if (THERMOSTAT_MODE.propertyName.equals(property)) {
+            return new ChannelInfo[] { THERMOSTAT_MODE };
+        }
         return null;
     }
 
     @Override
     public void updateChannels(String interfaceName, List<JsonObject> stateList, UpdateChannelResult result) {
-        QuantityType<Temperature> temperatureValue = null;
         for (JsonObject state : stateList) {
+            QuantityType<Temperature> temperatureValue = null;
+            StringType operationValue = null;
+            logger.debug("Updating {} with state: {}", interfaceName, state.toString());
             if (TARGET_SETPOINT.propertyName.equals(state.get("name").getAsString())) {
-                JsonObject value = state.get("value").getAsJsonObject();
                 // For groups take the first
                 if (temperatureValue == null) {
+                    JsonObject value = state.get("value").getAsJsonObject();
+                    float temperature = value.get("value").getAsFloat();
+                    String scale = value.get("scale").getAsString().toUpperCase();
+                    if ("CELSIUS".equals(scale)) {
+                        temperatureValue = new QuantityType<Temperature>(temperature, SIUnits.CELSIUS);
+                    } else {
+                        temperatureValue = new QuantityType<Temperature>(temperature, ImperialUnits.FAHRENHEIT);
+                    }
+                }
+                updateState(TARGET_SETPOINT.channelId, temperatureValue == null ? UnDefType.UNDEF : temperatureValue);
+            }
+            if (THERMOSTAT_MODE.propertyName.equals(state.get("name").getAsString())) {
+                // For groups take the first
+                String operation = state.get("value").getAsString().toUpperCase();
+                operationValue = new StringType(operation);
+                updateState(THERMOSTAT_MODE.channelId, operationValue == null ? UnDefType.UNDEF : operationValue);
```

Comment:
`operationValue` cannot be null, so either some logic is wrong, or this check is not needed.